Annotato/Annotato/ViewModels/Annotation/AnnotationViewModel.swift

```diff
@@ -278,11 +299,41 @@ extension AnnotationViewModel {
 
 extension AnnotationViewModel {
     func didDelete() {
+        guard !isResolving else {
+            return
+        }
         isRemoved = true
         selectionBox.didDelete()
         document?.removeAnnotation(annotation: self)
     }
 
+    func didSaveMergeConflicts() {
+        guard NetworkMonitor.shared.isConnected else {
+            AnnotatoLogger.error("Save merge conflict button pressed while offline")
+            return
+        }
+        Task {
+            await annotationsPersistenceManager.createOrUpdateAnnotation(annotation: model)
+        }
+        resolveBySave = true
+        isResolving = false
+    }
+
+    func didDiscardMergeConflicts() {
+        guard NetworkMonitor.shared.isConnected else {
+            AnnotatoLogger.error("Save merge conflict button pressed while offline")
+            return
+        }
+        selectionBox.didDelete()
+        document?.removeAnnotationWithoutPersistence(annotation: self)
+        model.setDeletedAt(to: Date())
+        Task {
+            await annotationsPersistenceManager.createOrUpdateAnnotation(annotation: model)
+        }
```

Comment:
I feel like this whole thing is very strange. It's kind of unintuitive at first glance because it's a discard function, but there's a removeWithoutPersistence followed by a setDeletedAt (which should be in the persistence layer), and then a createOrUpdate.

I think you can just call the normal delete path, to ensure everything happens properly?? Then in the backend deleteAnnotation method, just add logic to create it first if it does not exist. But if this is done, the view models receiveX stuff need to be worked out properly. Right now they're a bit off also, but unrelated to this PR.